### DISCUSSION TITLE
Docker development command not working

### DISCUSSION THREAD

**Discussion by ShivangRawat30** [learningequality/kolibri#11213] [Q&A]
When attempting to set up the project using Docker, I ran the command `make docker-build-base` and encountered the following error:

```
python -c "import kolibri; print(kolibri.__version__)" > kolibri/VERSION
/bin/bash: python: command not found
make: *** [writeversion] Error 127
```

OS: mac-m1
IDE: VSCode

**Comment by bjester** [comment]
Hi @ShivangRawat30,

Our development instructions for Docker need some attention, as I'm afraid they're not written with completely new development environments in mind. The `make docker-build-base` recipe assumes you already have python installed on your host and the kolibri project set up on your host as well.

Were you led to try the Docker approach because of issues with the host based setup? I see you're on an M1 Mac

**Comment by ShivangRawat30** [comment]
> Were you led to try the Docker approach because of issues with the host based setup? I see you're on an M1 Mac

I typically opt for setting up projects using Docker since it significantly reduces time, but since it's currently not functioning correctly, I'll attempt to set it up using a host-based approach.

**Comment by bjester** [comment]
Sorry for the inconvenience. I've added an issue for us to resolve the misleading documentation https://github.com/learningequality/kolibri/issues/11214

**Comment by ShivangRawat30** [comment]
@bjester I tried the command after installing python and now I am getting this error.

```
base.dockerfile:23
--------------------
  22 |     # install nodejs and yarn
  23 | >>> RUN apt-get update && \
  24 | >>>     curl -sSO https://deb.nodesource.com/node_16.x/pool/main/n/nodejs/nodejs_$NODE_VERSION-1nodesource1_amd64.deb && \
  25 | >>>     dpkg -i ./nodejs_$NODE_VERSION-1nodesource1_amd64.deb && \
  26 | >>>     rm nodejs_$NODE_VERSION-1nodesource1_amd64.deb && \
  27 | >>>     apt-get install yarn
  28 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c apt-get update &&     curl -sSO https://deb.nodesource.com/node_16.x/pool/main/n/nodejs/nodejs_$NODE_VERSION-1nodesource1_amd64.deb &&     dpkg -i ./nodejs_$NODE_VERSION-1nodesource1_amd64.deb &&     rm nodejs_$NODE_VERSION-1nodesource1_amd64.deb &&     apt-get install yarn" did not complete successfully: exit code: 1
```

**Comment by MisRob** [comment]
Hi @ShivangRawat30, thank you for your report. We're starting to work on some improvements and one of our team members will reach out again when we have some news.

Hopefully you were able to set Kolibri dev en without Docker meanwhile. Let us know if there were any problems with that.

**Comment by ShivangRawat30** [comment]
@MisRob  I'm attempting to set up my project and encountered an issue while running this command `pyenv activate kolibri-py3.9` The error message I received is as follows:

```
Failed to activate virtualenv.

Perhaps pyenv-virtualenv has not been loaded into your shell properly.
Please restart current shell and try again.
```

I have already attempted to restart the shell, but the problem persists. Additionally, I've already created a virtual environment using `pyenv virtualenv 3.9.9 kolibri-py3.9` and the terminal responded with:
```
pyenv-virtualenv: `/Users/shivangrawat/.pyenv/versions/kolibri-py3.9' already exists.

```

**Comment by MisRob** [comment]
Hi @ShivangRawat30, could you upload your logs from the very beginning of the setup?

**Reply by ShivangRawat30** [reply]
I've solved the issue, thank you for the help

**Comment by MisRob** [comment]
Leaving this discussion open as the original Docker issue is not resolved yet. cc @vkWeb

**Comment by MisRob** [comment]
Closing as we have some Docker setup refactor underway
